Sprint7 task4 name to employee

iAcay opened the pull request:
Hey! The code which is below was implemented in order to meet the acceptance criterias of the last task in sprint no. 7 :)

AC in this task:
-- On the registration form there are two new fields: first_name and last_name
-- An admin should be able to edit an employee's first & last name
-- Existing employees should be able to edit their profile
-- There is a presence validation set on first & last name
-- There is NOT NULL constraint set on employee's first & last name

In order to solve problem with NOT NULL constraint on existing employee's first & last name I decided to implement first migration which contains default values for each these columns. After that I did second migration for delete default values :)

https://user-images.githubusercontent.com/96024046/184517106-80a235b0-f5f5-43b0-acd3-6f6427a32cc7.mov

iAcay committed:
frontend: remove unnecessary </p> tags

iAcay committed:
feat: add first and last name to employees' table

-- first migration adds first and last name with default option for existing employees in order to solve problem with constraint null: false
-- second migration removes already unnecessary defaults values

iAcay committed:
feat: add first and last name to employees

-- add possibility for employees to sign up and edit account with additional first_name and last_name fields
-- add possibility for admin to edit employees' first & last name
-- add presence: true validation for first & last name
-- implement full_name method which returns email only when first & last name == 'No Name'
-- adjust views to introduction first & last name for employees -> if employee's full name != default 'No Name', other employees should recognize him by his full_name instead of email

iAcay committed:
tests: adjust tests for introduction employees' first & last name

maikhel commented on app/views/admin_users/employees/index.html.erb:
```diff
@@ -12,6 +13,7 @@
     <tbody>
       <% employees.each do |employee| %>
         <tr class="text-center">
+          <td><%= employee.first_name + ' ' + employee.last_name %></td>
```
You could use `employee.full_name` method here

iAcay replied:
First of all, thanks for CR! 👍 In index.html.erb file I decided to not use a full_name method because this method shows employee's email or first & last name. On this page I want to give admins both of these values. First column of the table contains first & last name wheras second column contains employee's email.

maikhel commented on app/controllers/AdminUsers/employees_controller.rb:
```diff
@@ -55,7 +55,7 @@ def employee
     end
```
Just detail, but in Rails all file directories and filenames are in snake case, so here you could have `admin_users`

iAcay replied:
Corrected :) Thanks!

maikhel commented on app/controllers/employees/parameter_sanitizer.rb:
```diff
@@ -0,0 +1,9 @@
+module Employees
+  class ParameterSanitizer < Devise::ParameterSanitizer
```
It's a little confusing to have non-controller class in `controllers` dir.

I think you could have a simpler solution with:
```
before_action :configure_permitted_parameters, if: :devise_controller?


def configure_permitted_parameters
    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:first_name, :last_name, :email, :password) }
    devise_parameter_sanitizer.permit(:account_update) do |u|
        u.permit(:first_name, :last_name, :email, :password, :current_password)
    end
end


```

iAcay replied:
I have changed it 💯

maikhel reviewed (approved):
Looks good! 👍

iAcay committed:
refactor: move devise configure_permitted_parameters to application controller

-- change folder name from AdminUsers to admin_users
-- remove parameter_sanitizer file and implement configure_permitted_parameters method in application controller
